packages/tools/src/stateManagement/segmentation/getViewportSegmentations.ts

```diff
@@ -0,0 +1,35 @@
+import type { SegmentationRepresentations } from '../../enums';
+import type { Segmentation } from '../../types';
+import { getSegmentation } from './getSegmentation';
+import { defaultSegmentationStateManager } from './SegmentationStateManager';
+
+/**
+ * Retrieves the viewport IDs that have a specific segmentation.
+ * @param segmentationId - The ID of the segmentation.
+ * @returns An array of viewport IDs that have the specified segmentation.
+ */
+export function getViewportSegmentations(
+  viewportId: string,
+  type: SegmentationRepresentations
+): Segmentation[] {
+  const segmentationStateManager = defaultSegmentationStateManager;
+  const state = segmentationStateManager.getState();
+  const viewports = state.viewports;
+
+  const segmentationsForViewportId = viewports[viewportId];
+
+  const segmentations = segmentationsForViewportId.map(
+    (segmentationInViewport) =>
+      getSegmentation(
+        type && segmentationInViewport.type === type
```

Comment:
If you are trying to allow excluding the type to mean any type, then this set of lines should be:
(!type || segmentationInViewport.type===type)